### ISSUE TITLE
How is GAN related to the generator in default training setting?

### ISSUE THREAD

**Issue by Owen-Liuyuxuan** [gruberto/Gated2Depth#3]
Many thanks for your effort and kindness to open source the project!

"--adv_weight" does not seem to be modified in any of the launching scripts and remained 0. Does it look like the generator network would not receive any loss gradients from the discriminator? 

Would like to know how GAN is related in the training process.

**Comment by gruberto** [comment]
In general, the adversarial weight is a hyperparameter you can play with. When training on synthetic data, you can train both generator and discriminator since dense ground truth depth is available. For training on real data, you cannot train the discriminator due to sparse ground truth depth from lidar. However, you can exploit the synthetic discriminator for an additional adversarial weight. For more details, we refer to our paper: https://arxiv.org/pdf/1902.04997.pdf
